paper.md

```diff
@@ -24,36 +24,47 @@ bibliography: paper.bib
 
 # Summary
 
-The tensor framework JAX [@jax] shows excellent performance on both machine learning and scientific computing workloads, while all user code is written in pure Python.
+The tensor framework JAX [@jax] combines expressivity and performance while retaining an accessible pure Python interface.
+Expressivity is achieved by treating functions as first-class objects, while efficiency is obtained by compiling to machine code Just-Ahead-Of-Time.
```

Comment:
> Expressivity is achieved by treating functions as first-class objects

It is more than that, isn't it? I called it "functional design" which for me encompasses first-class functions + higher-order functions + pure functions.

Reply:
It's about jargon, again, i think.
First-class objects means you can operate on them as you would operate on anything else.

I personally think that `higher order functions` it's somewhat an outdated concept. If you can pass a function as an argument, then you automatically have those.

If they are first-class objects of course you can pass them to functions.
But i'm not a computer scientist...

Pure functions is unrelated and it's simply a tool for helping the optimiser.

Reply:
> Pure functions is unrelated and it's simply a tool for helping the optimiser.

It's also increasing expressivity, isn't it? Because you don't need to worry about internal state.

Anyhow, how about a compromise: We keep this, but change JAOT to JIT? (I agree that JIT might not be technically correct, but it's what the community is familiar with.)

Reply:
Whatever, JAOT is OK, too.